sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureDefaultTokenCredentialAutoConfiguration.java

```diff
@@ -39,8 +44,11 @@ public TokenCredential azureTokenCredential(AzureCredentialBuilderFactory<Defaul
     }
 
     @Bean
-    public AzureCredentialBuilderFactory<DefaultAzureCredentialBuilder> defaultAzureCredentialBuilderFactory() {
-        return new AzureCredentialBuilderFactory<>(azureGlobalProperties, new DefaultAzureCredentialBuilder());
+    public AzureCredentialBuilderFactory<DefaultAzureCredentialBuilder> defaultAzureCredentialBuilderFactory(
+        @Autowired(required = false) ThreadPoolTaskExecutor taskExecutor) {
```

Comment:
Does this mean we are going to autowire any ThreadPoolTaskExecutor?

Reply:
Yes, the `ThreadPoolTaskExecutor` is maintained by Spring boot.